src/adapter/src/catalog.rs

```diff
@@ -2037,6 +2037,23 @@ impl CatalogEntry {
         matches!(self.item(), CatalogItem::Log(_))
     }
 
+    /// Reports whether this catalog entry can be treated as a relation, it can produce rows.
+    pub fn is_relation(&self) -> bool {
+        match self.item {
+            CatalogItem::Table(_)
+            | CatalogItem::Source(_)
+            | CatalogItem::Log(_)
+            | CatalogItem::View(_)
+            | CatalogItem::MaterializedView(_)
+            | CatalogItem::Index(_) => true,
```

Comment:
Should this be true for indexes? In other similar places (e.g. `desc()`) we return `false` for indexes.

If it's ok to return `false` for indexes, perhaps simplify to `.desc().is_ok()`?

Reply:
What does `depend_on` contain in `sequenc_plan_view`? Is it the views/tables/sources over which the view being created depends, or is it the index ids produced by the optimizer? That'd affect the ability to simplify as above.

Reply:
You're right, indexes should be `false`. I didn't use `desc()` though because it takes a `FullObjectName` which we don't have access to.

Reply:
> What does depend_on contain in sequence_plan_view? Is it the views/tables/sources over which the view being created depends, or is it the index ids produced by the optimizer? That'd affect the ability to simplify as above.

It's the first "the views/tables/sources over which the view being created depends". It's what is returned by the `NameResolver`.